AdminServer/appscale/admin/utils.py

```diff
@@ -581,3 +591,72 @@ def _constant_time_compare(val_a, val_b):
   constant_time_compare = hmac.compare_digest
 else:
   constant_time_compare = _constant_time_compare
+
+
+def validate_routing_rule(rule, services):
+  """
+  domain: string. Domain name to match against. The wildcard "*" is supported
+    if specified before a period: "*.". Defaults to matching all domains: "*".
+  path: string. Pathname within the host. Must start with a "/". A single "*"
+    can be included at the end of the path. The sum of the lengths of the
+    domain and path may not exceed 100 characters.
+  service: string. Resource ID of a service in this application that should
+    serve the matched request. The service must already exist. Example: default.
+  Tip: You can include glob patterns like the `*` wildcard character in the
+   `url` element; however, those patterns can be used only before the host name
+   and at the end of the URL path.
+
+  """
+  service = rule['service']
+  domain = rule['domain']
+  path = rule['path']
+  if service not in services:
+    raise InvalidDispatchConfiguration('Service does not exist.')
+  if domain.startswith('*'):
+    if not _URL_HOST_SUFFIX_PATTERN_RE.match(domain):
+      raise InvalidDispatchConfiguration(
+        'Invalid host pattern {}'.format(domain))
+  else:
+    if not _URL_HOST_EXACT_PATTERN_RE.match(domain):
+      raise InvalidDispatchConfiguration(
+          'Invalid host pattern {}'.format(domain))
+    matcher = _URL_IP_V4_ADDR_RE.match(domain)
+    if matcher and sum(1 for x in matcher.groups() if int(x) <= 255) == 4:
+      raise InvalidDispatchConfiguration(
+          'Host may not match an ipv4 address {}'.format(domain))
+
+  if not DISPATCH_PATH_REGEX.match(path):
+    raise InvalidDispatchConfiguration('Invalid path pattern {}.'.format(path))
+
+  if not DISPATCH_DOMAIN_REGEX_SINGLE_ASTERISK.match(domain):
+    asterisks = DISPATCH_DOMAIN_REGEX_ASTERISKS.findall(domain)
+    asterisk_dot = DISPATCH_DOMAIN_REGEX_ASTERISK_DOT.findall(domain)
+    if len(asterisks) != len(asterisk_dot):
+      raise InvalidDispatchConfiguration(
+          'Invalid host pattern {}'.format(domain))
+
+  if len(domain) + len(path) > 100:
+    raise InvalidDispatchConfiguration('URL over 100 characters.')
+
+
+def routing_rules_from_dict(payload, project_id, services):
+  try:
+    given_routing_rules = payload['dispatchRules']
+  except KeyError:
+    raise InvalidQueueConfiguration('Payload must contain dispatchRules')
+
+  rules = []
+  for rule in given_routing_rules:
+    validate_routing_rule(rule, services)
+    rules.append(convert_to_nginx_friendly_rule(rule, project_id))
```

Comment:
Though the validation is fine, the Admin API should store the configuration as given. The nginx implementation modifications should happen by whatever is writing the nginx config file.